_source/_posts/2021-09-12-spring-boot-spa.md

```diff
@@ -0,0 +1,686 @@
+
+Select version `3.x`. 
+
+Enable `history mode for router` (type `y`).
+
+Select `ESLint with error prevention only`.
+
+The defaults for the remainder of the options are fine: lint on save, saving configuration in dedicated files, and not saving this as a preset.
+
+Navigate into the project directory.
+
+```bash
+cd okta-app
+```
+
+You need to install a few dependencies. The first three are required for Okta security and the sign-in widget. The last, `axios`, is the library you'll use to make the HTTP request to the resource server.
+
+```bash
+npm install --save \
+@okta/okta-signin-widget \
+@okta/okta-vue \
+@okta/okta-auth-js \
+axios
+```
+
+Create a `src/okta/index.js` file:
+
+```vue
+import OktaSignIn from '@okta/okta-signin-widget'
+import { OktaAuth } from '@okta/okta-auth-js'
+
+const yourOktaUri = '${yourOktaUri}';
+const clientId = '${clientId}';
+
+const oktaSignIn = new OktaSignIn({
+    baseUrl: `${yourOktaUri}`,
+    clientId: clientId,
+    redirectUri: 'http://localhost:8080/login/callback',
+    authParams: {
+        pkce: true,
+        issuer: `${yourOktaUri}/oauth2/default`,
+        display: 'page',
+        scopes: ['openid', 'profile', 'email']
+    }
+});
+
+const oktaAuth = new OktaAuth({
+    issuer: `${yourOktaUri}/oauth2/default`,
+    clientId: clientId,
+    redirectUri: window.location.origin + '/login/callback',
+    scopes: ['openid', 'profile', 'email']
+})
+
+export { oktaAuth, oktaSignIn };
+```
+
+In the file above, **you need to replace the placeholders `${...}` with the appropriate values**. Before you can do this, however, you need to use the CLI to create the OIDC application on the Okta servers, which you will do directly below.
+
+## Create A Single-Page OIDC Application On Okta
+
+This time you're going to create a single-page application for the front-end (instead of the web application type you used for the resource server). Enter the following command into a shell.
+
+```bash
+okta apps create
+```
+
+Press `enter` to accept the default **Application name** `okta-app`. (Application name here refers to the OIDC application created on the Okta servers).
+
+For **Type of Application**, Select `2: Single Page App`.
+
+Change the **Redirect URI** to `http://localhost:8080/login/callback`.
+
+Press `enter` to accept the default **Post Logout Redirect**.
+
+You should see some output like the following.
+
+```bash
+Okta application configuration: 
+Issuer:    https://{yourOktaUri}/oauth2/default
+Client ID: {clientId}
+```
+
+The values in brackets will, of course, be your actual values. 
+
+**You need to go back to the `src/okta/index.js` file and replace the `yourOktaUri` and `clientId` variables at the top of the file with your values.** The value for `yourOktaUri` is the Okta URI without any further path specifiers. It will look like this: `https://dev-123456.okta.com`
```

Comment:
```suggestion
**You need to go back to the `src/okta/index.js` file and replace the `yourOktaUri` and `clientId` variables at the top of the file with your values.** The value for `yourOktaUri` is the Okta URI without any further path specifiers. It will look like this: `https://dev-133337.okta.com`
```
consistency